src/components/articles/load-more-button.tsx

```diff
@@ -0,0 +1,36 @@
+"use client";
+
+import React, { useState } from "react";
+
+import { ArticleCard } from "@/components/articles/article-card";
+
+const LoadMoreArticles = ({ username }: { username: string }) => {
+  const [articles, setArticles] = useState<any[]>([]);
+  const [offset, setOffset] = useState(10);
+  const [hasMore, setHasMore] = useState(false);
+
+  const loadArticles = async () => {
+    //this should be /api/articles?
+    const response = await fetch(
+      `/api/article?username=${username}&limit=10&offset=${offset}`,
+    );
+    if (response.ok) {
+      const newArticles = await response.json();
+      setArticles((prev) => [...prev, ...newArticles]);
+      setOffset((prev) => prev + newArticles.length);
+
+      setHasMore(newArticles.length === 10);
+    }
+  };
+
+  return (
+    <>
+      {articles.map((article) => (
+        <ArticleCard key={article.id} article={article} />
+      ))}
```

Comment:
Shouldn't this be wrap around some <div>. Then we can do `flex flex-col gap-4`. We can use gap to control the spacing between each article.

Reply:
added wrapper at src\app\(public)\[username]\page.tsx